examples/nlp/token_classification/punctuate_capitalize.py

```diff
@@ -0,0 +1,187 @@
+    output = parser.add_mutually_exclusive_group(required=True)
+    output.add_argument(
+        "--output_manifest",
+        "-M",
+        type=Path,
+        help="Path to output NeMo manifest. Text with punctuation and capitalization will be under 'pred_text' key "
+        "if 'pred_text' key is present in the first element of input manifest. Otherwise text with restored "
+        "punctuation and capitalization will be under 'text' key. Exactly one parameter of `--output_manifest` and "
+        "`--output_text` should be provided.",
+    )
+    output.add_argument(
+        "--output_text",
+        "-T",
+        type=Path,
+        help="Path to file with text with restored punctuation and capitalization. Exactly one parameter of "
+        "`--output_manifest` and `--output_text` should be provided.",
+    )
+    model = parser.add_mutually_exclusive_group(required=False)
+    model.add_argument(
+        "--pretrained_name",
+        "-p",
+        help="The name of NGC pretrained model. No more than one of parameters `--pretrained_name`, `--model_path`"
+        "should be provided.",
+        choices=[m.pretrained_model_name for m in PunctuationCapitalizationModel.list_available_models()],
+    )
+    model.add_argument(
+        "--model_path",
+        "-P",
+        type=Path,
+        help="Path to .nemo checkpoint of punctuation and capitalization model. No more than one of parameters "
+        "`--pretrained_name` and `--model_path` should be provided.",
+    )
+    parser.add_argument(
+        "--max_seq_length",
+        "-L",
+        type=int,
+        default=64,
+        help="Maximum length of segments into which queries are split. `max_seq_length` includes [CLS] and [SEP] "
+        "tokens.",
+    )
+    parser.add_argument(
+        "--margin",
+        "-g",
+        type=int,
+        default=16,
+        help="number of subtokens in the beginning and the end of segments which are not used for prediction "
+        "computation. The first segment does not have left margin and the last segment does not have right margin. For "
+        "example, if input sequence is tokenized into characters, ``max_seq_length=5``, ``step=1``, and ``margin=1`` "
+        "than query 'hello' will be tokenized into segments ``[['[CLS]', 'h', 'e', 'l', '[SEP]'], ['[CLS]', 'e', 'l', "
+        "'l', '[SEP]'], ['[CLS]', 'l', 'l', 'o', '[SEP]']]``. These segments are passed to the model. Before final "
+        "predictions computation, margins are removed. In the next list, subtokens which logits are not used for final "
+        "predictions computation are marked with asterisk: ``[['[CLS]'*, 'h', 'e', 'l'*, '[SEP]'*], ['[CLS]'*, 'e'*, "
+        "'l', 'l'*, '[SEP]'*], ['[CLS]'*, 'l'*, 'l', 'o', '[SEP]'*]]``.",
+    )
+    parser.add_argument(
+        "--step",
+        "-s",
+        type=int,
+        default=8,
+        help="Relative shift of consequent segments into which long queries are split. Long queries are split into "
```

Comment:
Are the defaults set here recommended values?

Reply:
I written to the script description that this values are optimal for IWSLT 2019 test dataset.